samples/ipc_call_interface/inc/math_client.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#include "az_ulib_ipc_api.h"
+#include "az_ulib_result.h"
+#include "math.h"
+#include <stdint.h>
+
+static inline az_ulib_result math_create(math_handle handle) {
+  return az_ulib_ipc_try_get_interface(
+      MATH_INTERFACE_NAME, MATH_INTERFACE_VERSION, AZ_ULIB_VERSION_EQUALS_TO, handle);
+}
+
+static inline void math_destroy(math_handle handle) { az_ulib_ipc_release_interface(handle); }
+
+/*
+ * Inter-component Calling Wrapper for sum.
+ */
+static inline az_ulib_result math_sum(math_handle handle, int32_t a, int32_t b, int64_t* res) {
```

Comment:
this is the caller's implementation of the ipc_call, correct? I think it's a tad confusing to name the caller's functions with "math" when "math" describes the interface. Maybe rename to something like "client_work" so the sample shows how the client is calling the math interface to do work.

Reply:
This is autogenerated from the metadata, and it is specific for the "class" math. In COM, we would call this function RCW [Runtime Callable Wrapper], in our case something like AZCW [Azure Callable Wrapper], and the signature for .NET would be. 
```c#
public class math {
  public int64_t sum(int32_t a, int32_t b) { /* marshalling, call, unmarshalling */ }
  public int64_t subtract(int32_t a, int32_t b) { /* marshalling, call, unmarshalling */ }
}
```
Perhaps we should create a "wrappers" directory for that as well.

---
In reply to: [529831417](https://github.com/Azure/azure-ulib-c/pull/53#discussion_r529831417) [](ancestors = 529831417)